src/auth/__tests__/authService.test.js

```diff
@@ -269,18 +283,87 @@ describe('authService', () => {
     it('should call sessionStorage.setItem with the right arguments', async () => {
       const setSpy = jest.spyOn(Storage.prototype, 'setItem');
       expect.assertions(2);
-      await authService.fetchApiToken(mockUser);
+      await authService.fetchAndStoreApiToken(mockUser);
 
       expect(setSpy).toHaveBeenCalledTimes(1);
       expect(setSpy.mock.calls[0]).toMatchInlineSnapshot(`
         Array [
           "apiToken",
-          "71ffd52c-5985-46d3-b445-490554f4012a",
+          "${apiToken}",
         ]
       `);
     });
   });
 
+  describe(`Api tokens are fetched again after user tokens are renewed.
+            After silent renew completes, the _userLoaded event is raised and`, () => {
+    beforeEach(() => {
+      global.fetch.resetMocks();
+      mockFetchApiToken();
+      jest.useFakeTimers();
+    });
+
+    afterEach(() => {
+      jest.useRealTimers();
+    });
+    beforeAll(() => {
+      enableActualHttpPoller(jest.requireActual('../http-poller'));
+    });
+
+    it('it calls fetchApiToken and stores new token after success', async () => {
+      const { sessionUser } = setSession({
+        validUser: true,
+        validApiToken: true,
+      });
+      sessionStorage.setItem(API_TOKEN, 'foobar');
+
+      const fetcApiTokenSpy = jest.spyOn(authService, 'fetchApiToken');
+
+      await userManager.events._userLoaded.raise(sessionUser);
+      expect(fetcApiTokenSpy).toHaveBeenCalledTimes(1);
+      jest.advanceTimersByTime(501);
```

Comment:
Where is this magic number coming from (it's present more than once in this file)? I'm guessing there's probably a value 500 somewhere and this is supposed to be more than that. So perhaps that magic number 500 can be used here and then +1 to it. So something like: `theUltimateDelay + 1`.

Reply:
Added describing constant for this